projects/ngx-sub-form/src/lib/ngx-root-form.component.ts

```diff
@@ -59,6 +60,12 @@ export abstract class NgxRootFormComponent<ControlInterface, FormInterface = Con
       .subscribe();
   }
 
+  // This method will allow for a smart component to reset the dumb form.
+  public reset() {
+    const nullVal = JSON.parse(JSON.stringify(this.formGroup.value).replace(/(?<=:\s?)(?:(["']).*?(?<!\\)\1|\w+|[.0-9]+)(?=[, \]}])/g, 'null'));
+    this.formGroup.reset(nullVal);
```

Comment:
Is the purpose of that to set every value of the object to `null`?
You could just iterate over `formGroupControls` using `Object.keys` and set that to null.

But the issue is that if we're using within the formGroup default values they'd be lost.
We should probably just do what the `reset` method is doing and let the consumer of the lib pass the argument to reset (and it could be `FormInterface | undefined`).

Reply:
thank you maxime. The purpose is transform

```
{
  "personalDetails": {
    "name": {
      "firstname": "randomString",
      "lastname": "randomString"
    },
    "gender": "Male",
    "address": {
      "streetaddress": "randomString",
      "city": "randomString",
      "state": "randomString",
      "zip": "randomString",
      "country": "randomString"
    },
    "phone": {
      "phone": "randomString",
      "countrycode": "randomString"
    }
  }
}
```

into

(*object111)
```
{
  "personalDetails": {
    "name": {
      "firstname": null,
      "lastname": null
    },
    "gender": null,
    "address": {
      "streetaddress": null,
      "city": null,
      "state": null,
      "zip": null,
      "country": null
    },
    "phone": {
      "phone": null,
      "countrycode": null
    }
  }
}
```

and use the latter as reset value in RootFormComponent.

If I would loop over every control and set it to null this would trigger formGroup.reset() in every subComponent, which im trying to avoid.

it would transform for e.g.

```
"phone": {
      "phone": "randomString",
      "countrycode": "randomString"
}
```

into

`"phone": null`

which triggers reset in phone-component. I guess its good to have one reset() in RootForm by setting (*object111) as reset.value.

Lets see how it works out. Hope im clear :-).

btw my regex (currently) would not work when I have an array of primives.